src/mailer/engine.py

```diff
@@ -226,7 +235,8 @@ def send_all():
         release_lock(lock)
 
     logging.info("")
-    logging.info("%s sent; %s deferred;" % (sent, deferred))
+    for action_taken, cnt in counts.items():
```

Comment:
Please add a `sorted(..)` wrap here for deterministic behaviour with Python prior to 3.7 where dictionaries are not ordered by default.